pkg/validation/app.go

```diff
@@ -397,7 +397,10 @@ func (v *Validator) validateUserConfig(ctx context.Context, cr v1alpha1.App) err
 		}
 
 		_, err := v.k8sClient.CoreV1().ConfigMaps(ns).Get(ctx, key.UserConfigMapName(cr), metav1.GetOptions{})
-		if apierrors.IsNotFound(err) {
+		if key.IsManagedByFlux(cr, v.projectName) {
```

Comment:
I thought about this some more and I think a generic method that checks if a configmap or secret exists would be useful. We should also have this logic for the kubeconfig secret and the CM and secret in `.spec.config`.

The other thing I think would be useful is a flag for whether `IsManagedByFlux` is active. We want to allow the app CR to be created in app-admission-controller but prevent it being processed in the `validation` resource in app-operator until all the CMs and secrets exist.

WDYT?

Reply:
I believe you are right about the generic method, since we have the same checks in the `validateConfig` it could be usefull to export it to a separate function and not repeat the code. Will try that.

And could you elaborate more on the flag for whether `IsManagedByFlux` is active? Do you mean an extra boolean argument to the `Validate()` function, so that we can set it `true` in the `app-admission-controller`, but `false` in the `app-operator`?

Reply:
My idea is to add the flag as a setting for the Validator [here](https://github.com/giantswarm/app/blob/15abc12a14703c87048341693f195d51fec15f0b/pkg/validation/validator.go#L15-L16).

So we only need to set it once when the validator is configured.

Reply:
ah, makes total sense, let me try that as well

Reply:
@rossf7 wdyt now?